.github/workflows/build_and_test.yml

```diff
@@ -36,13 +36,14 @@ jobs:
       - name: Checkout repository
         uses: actions/checkout@v3
       - name: Set up JDK ${{ matrix.java-version }}
-        uses: actions/setup-java@v2
+        uses: actions/setup-java@v4
         with:
           java-version: ${{ matrix.java-version }}
           distribution: 'adopt'
+          cache: 'gradle'
       - name: Build with Gradle
         run: |
-          set -o pipefail; ./gradlew build; set +o pipefail
```

Comment:
I removed `pipefail` here. If we need this, we should put it inside `gradlew` file.